src/botPage/view/View.js

```diff
@@ -467,8 +468,19 @@ export default class View {
         });
     }
     stop() {
+        try {
+            const { contractId } = this.blockly.interpreter.bot.tradeEngine;
+            if (contractId) {
+                api.subscribeToOpenContract(contractId);
+            }
+            api.events.on('proposal_open_contract', r => {
+                const contract = r.proposal_open_contract;
+                broadcastContract({ accountID: $('.account-id').html(), ...contract });
```

Comment:
Better to add a check for `is_sold` or `sell_price` to unsubscribe from the open contact when the contract is sold.

Reply:
Also better to unsubscribe this when starting a new bot instance. I'm not sure if we show a correct trade table if two unrelated contracts are broadcasted.